observability-test/observability.test.ts

```diff
@@ -0,0 +1,417 @@
+      sql: req,
+    };
+
+    startTrace('aSpan', opts, async span => {
+      assert.equal(
+        span.attributes[SEMATTRS_DB_STATEMENT],
+        undefined,
+        'Mismatched DB_STATEMENT attribute'
+      );
+    });
+  });
+
+  it('alwaysOffSampler used, no spans exported', () => {
+    const exporter2 = new InMemorySpanExporter();
+    const provider2 = new NodeTracerProvider({
+      sampler: new AlwaysOffSampler(),
+      exporter: exporter2,
+    });
+    provider2.addSpanProcessor(new SimpleSpanProcessor(exporter2));
+    provider2.register();
+
+    startTrace('aSpan', {tracerProvider: provider2}, async span => {
+      await new Promise((resolve, reject) => setTimeout(resolve, 400));
+      span.end();
+
+      const gotSpansFromGlobal = exporter.getFinishedSpans();
+      assert.ok(gotSpansFromGlobal.length, 0, 'no spans should be exported');
+
+      const gotSpansFromCurrent = exporter2.getFinishedSpans();
+      assert.ok(
+        gotSpansFromCurrent.length,
+        0,
+        'no span should have been exported'
+      );
+
+      exporter2.forceFlush();
+      await provider2.shutdown();
+    });
+  });
+});
+
+describe('getActiveOrNoopSpan', () => {
+  it('getActiveOrNoopSpan with no value should return a noopSpan and nothing exported', () => {
+    const span = getActiveOrNoopSpan();
+    assert.ok(span, null, 'the span MUST not be null regardless');
+    span.updateName('aSpan should not crash');
+    span.setStatus({message: 'done here'});
+  });
+
+  it('with a started span should return the currently active one', () => {
+    const exporter = new InMemorySpanExporter();
+    const provider = new NodeTracerProvider({
+      sampler: new AlwaysOffSampler(),
+      exporter: exporter,
+    });
+    provider.addSpanProcessor(new SimpleSpanProcessor(exporter));
+    provider.register();
+
+    startTrace('aSpan', {tracerProvider: provider}, async span => {
+      const activeSpan = getActiveOrNoopSpan();
+      assert.strictEqual(span.name, 'aSpan', 'names must match');
+      assert.strictEqual(
+        span.name,
+        activeSpan.name,
+        'names must match between activeSpan or current one'
+      );
+      assert.strictEqual(
+        span.startTime,
+        activeSpan.startTime,
+        'startTimes must match'
+      );
+      assert.strictEqual(
+        span.duration(),
+        undefined,
+        'the unended span must have an undefined duration'
+      );
+      assert.strictEqual(
+        activeSpan.duration(),
+        undefined,
+        'the unended span must have an undefined duration'
+      );
+      assert.strictEqual(
+        span.duration(),
+        activeSpan.duration(),
+        'durations must match'
+      );
+      span.end();
+      assert.strictEqual(
+        span.duration(),
+        activeSpan.duration(),
+        'durations must now match'
+      );
+    });
+  });
+});
+
+describe('setError', () => {
+  const exporter = new InMemorySpanExporter();
```

Comment:
Well so the use of "before" in majority of these cases unnecessarily adds typing declarations because the common code must be cleaned up in the after clause so that it won't pollute the other individual tests.